decidim-core/app/cells/decidim/content_blocks/base_cell.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ContentBlocks
+    # Base cell to wrap each content block which identifies also the resource
+    # the block belongs to
+    class BaseCell < Decidim::ViewModel
+      def resource
+        @resource ||= base_model.presence && base_model.find(model.scoped_resource_id)
+      end
+
+      private
+
+      def base_model
+        @base_model ||= options[:base_model] ||
+                        case model.scope_name
+                        when "participatory_process_group_homepage"
+                          Decidim::ParticipatoryProcessGroup
```

Comment:
Unfortunately the association of content blocks with different resources is not a typical polymorphic association where a pair of `scope_type` and `scope_id` could store the association with a participatory process, participatory process group or assembly (assemblies also will have a landing page composed of content blocks). In this way the way to find the resource would be transparent.

This method translates the name of scopes to associations with models. I agree that this is not a good solution, but it's necessary for example for decidim-assemblies/app/cells/decidim/assemblies/content_blocks/related_assemblies_cell.rb and decidim-participatory_processes/app/cells/decidim/participatory_processes/content_blocks/related_processes_cell.rb to determine the resource

What I can do is employ strings instead of calling classes directly and use safe_constantize, something like this:

```ruby
    class BaseCell < Decidim::ViewModel
      SCOPE_ASSOCIATIONS = {
        homepage: "Decidim::Organization",
        participatory_process_group_homepage: "Decidim::ParticipatoryProcessGroup",
        participatory_process_homepage: "Decidim::ParticipatoryProcess"
      }.freeze
      # [...]
      private

      def scope_associations
        SCOPE_ASSOCIATIONS.with_indifferent_access
      end

      def base_model
        @base_model ||= options[:base_model] || scope_associations[model.scope_name]&.safe_constantize
      end
      # [...]
```